rpcs3/Emu/NP/rpcn_client.cpp

```diff
@@ -918,14 +945,38 @@ namespace rpcn
 		avatar_url  = reply.get_string(false);
 		user_id     = reply.get<s64>();
 
-		auto get_usernames_and_status = [](vec_stream& stream, std::map<std::string, std::pair<bool, u64>>& friends)
+		auto get_usernames_and_status = [](vec_stream& stream, std::map<std::string, friend_online_data>& friends)
 		{
 			u32 num_friends = stream.get<u32>();
 			for (u32 i = 0; i < num_friends; i++)
 			{
 				std::string friend_name = stream.get_string(false);
-				bool online             = !!(stream.get<u8>());
-				friends.insert(std::make_pair(std::move(friend_name), std::make_pair(online, 0)));
+				bool online = !!(stream.get<u8>());
+
+				auto truncate_string = [](std::string& str, usz max_size)
```

Comment:
considering that this function exists twice, you should probably move this to fmt::truncate.
Also, the `if (>=) resize(size-1)` logic can simply be written as `if (>) resize(size)`